modules/storages/app/workers/storages/copy_project_folders_job.rb

```diff
@@ -30,13 +30,15 @@
 
 module Storages
   class CopyProjectFoldersJob < ApplicationJob
-    retry_on Errors::PollingRequired, wait: 3, attempts: (Rails.env.test? ? 3 : :unlimited)
+    include GoodJob::ActiveJobExtensions::Batches
+
+    retry_on Errors::PollingRequired, wait: 3, attempts: :unlimited
```

Comment:
What would be a good backoff algo here?

Reply:
It looks great but I'd like to cap the max time. In this case we should probably cap it at around one hour. 

Why? This is the time that MS is taking copying the files/folders around. It should be pretty fast so maybe, a slower - and capped - growth rate might be better. So maybe reducing the growth factor a bit and increasing the number of tries could be better, idk.

Reply:
I've played around with different factors:

A factor of 4 gives us 15 retries in ~2 days
A factor of 3 gives us 30 retries in ~2 days
A factor of 2 gives us 80 retries in ~2 days

I'm tending towards a factor closer to 2, so we check more often and fail around the same time.

PS: I do think that 2 days is too much btw. ~12 hours should be the cap until we get yelled at.

Reply:
Code that I used to verify this;

```ruby
backoff_lambda = ->(jitter, factor, i) do
  (i**factor) + (Kernel.rand * (i**factor) * jitter) + 2
end.curry

presentation_lambda = ->(list) do
  list.each_with_index do |duration, index|
    puts "Execution: #{index + 1} - Wait Time: #{ActiveSupport::Duration.build(duration).inspect} - Total Elapsed time: #{ActiveSupport::Duration.build(list[0..index].sum).inspect}"
  end
end

factor4 = backoff_lambda.(0, 4)
factor3 = backoff_lambda.(0, 3)
factor2 = backoff_lambda.(0, 2)

puts "Factor 4"
presentation_lambda[1.upto(15).map(&factor4)]

puts "Factor 3"
presentation_lambda[1.upto(30).map(&factor3)]

puts "Factor 2"
presentation_lambda[1.upto(80).map(&factor2)]
```

Reply:
When the job exhausts the number or retries it gets discarded (which should be logged). We could add the information about it to the email that gets send in the end.